x/gamm/pool-models/balancer/balancer_pool.go

```diff
@@ -380,52 +380,62 @@ func (pa *Pool) updateAllWeights(newWeights []PoolAsset) {
 // PokeTokenWeights checks to see if the pool's token weights need to be updated,
 // and if so, does so.
 func (pa *Pool) PokeTokenWeights(blockTime time.Time) {
-	// Pool weights aren't changing, do nothing.
-	poolWeightsChanging := (pa.PoolParams.SmoothWeightChangeParams != nil)
+	// check if pool weights didn't change
+	poolWeightsChanging := pa.PoolParams.SmoothWeightChangeParams != nil
 	if !poolWeightsChanging {
 		return
 	}
-	// Pool weights are changing.
+
 	// TODO: Add intra-block cache check that we haven't already poked
```

Comment:
@ValarDragon do we have an issue for this? If not, I'll create one.

Reply:
We don't have an issue for this, can you create one! Its super low priority, since it would only be of impact if we had multiple read only requests to a pool in the state machine. At the moment, we have no state machine method that just reads from a pool, with no writes

Reply:
Got it. Removed!